py/picca/io.py

```diff
@@ -1303,6 +1510,179 @@ def read_from_desi(nside,
+        z_qso: array of float
+            Redshift of the quasars
+        plate: array of integer
+            Plate number of the observations
+        mjd: array of integer
+            Modified Julian Date of the observations
+        fiberid: array of integer
+            Fiberid of the observations
+        order: 0 or 1 - default: 1
+            Order of the log10(lambda) polynomial for the continuum fit
+        pk1d: str or None - default: None
+            Format for Pk 1D: Pk1D
+
+    Returns:
+        List of read spectra for all the healpixs
+    """
+
+    spectra_in = glob.glob(os.path.join(in_dir,"**/coadd-*.fits"),recursive=True)
+    spectra = []
+    plate_unique=np.unique(plate)
+    for s in spectra_in:
+        for p in plate_unique:
+            if str(p)[:-1] in s:
+                spectra.append(s)
+                break
+
+    #tiles = [spectra[i].split("/")[-2].strip() for i in range(len(spectra))]
+    nights = []
+    data = {}
+    ztable = {t:z for t,z in zip(thingid,z_qso)}
+    ndata = 0
+
+    for i,spec in enumerate(spectra):
+        print("\rread tile {} of {}. ndata: {}".format(i,len(spectra),ndata))
+        try:
+            h = fitsio.FITS(spec)
+        except IOError:
+            print("Error reading pix {}\n".format(spec))
+            continue
+
+        if 'TARGET_RA' in h["FIBERMAP"].get_colnames():
+            ra = h["FIBERMAP"]["TARGET_RA"][:]*sp.pi/180.
+            de = h["FIBERMAP"]["TARGET_DEC"][:]*sp.pi/180.
+        elif 'RA_TARGET' in h["FIBERMAP"].get_colnames():
+            ra = h["FIBERMAP"]["RA_TARGET"][:]*sp.pi/180.
+            de = h["FIBERMAP"]["DEC_TARGET"][:]*sp.pi/180.
+            
+        petal_spec=h["FIBERMAP"]["PETAL_LOC"][:][0]
+        
+        if 'TILEID' in h["FIBERMAP"].get_colnames():
+            tile_spec=h["FIBERMAP"]["TILEID"][:][0]
+        else:
+            tile_spec=spec.split('-')[-2]    #minisv tiles don't have this in the fibermap
+
+        if 'NIGHT' in h["FIBERMAP"].get_colnames():
+            night_spec=h["FIBERMAP"]["NIGHT"][:][0]
+        else:
+            night_spec=int(spec.split('-')[-1].split('.')[0])
+        
+        
+        in_tids = h["FIBERMAP"]["TARGETID"][:]
+
+        specData = {}
+        try:
+            bands=['BRZ']
+            h['{}_WAVELENGTH'.format(bands[0])]
+            if i==0:
+                print("reading all-band coadd as in minisv dataset")
+        except (KeyError,IOError):
+            if i==0:
+                print("couldn't read the all band-coadd, trying single band as introduced in Andes reduction")
+            bands=['B','R','Z']
+
+        for str_band in bands:
+            dic={}
+            dic['LL'] = sp.log10(h['{}_WAVELENGTH'.format(str_band)].read())
+            dic['FL'] = h['{}_FLUX'.format(str_band)].read()
+            dic['IV'] = h['{}_IVAR'.format(str_band)].read()*(h['{}_MASK'.format(str_band)].read()==0)
+            w = sp.isnan(dic['FL']) | sp.isnan(dic['IV'])
```

Comment:
use `np`

Reply:
yep, never got why picca used sp. anyway for those things